proposed/building-arch-specific-applications.md

```diff
@@ -0,0 +1,806 @@
+
+As a result, the current behavior is considered to be unfortunate. It doesn't seem like the right behavior for developers intentionally targeting multiple architectures or for targeting single architectures.
+
+An example of the current default experience follows, with an application that has a single NuGet dependency (from Microsoft):
+
+```console
+C:\git\testapps\cpumath>type cpumath.csproj
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.ML.CpuMath" Version="0.11.0" />
+  </ItemGroup>
+
+</Project>
+
+C:\git\testapps\cpumath>dotnet build
+
+C:\git\testapps\cpumath>dir /s /b bin\*.dll bin\*.exe bin\*.so bin\*.dylib
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\cpumath.dll
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\Microsoft.ML.CpuMath.dll
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\cpumath.exe
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\runtimes\linux-x64\native\libCpuMathNative.so
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\runtimes\osx-x64\native\libCpuMathNative.dylib
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\runtimes\win-x64\native\CpuMathNative.dll
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\runtimes\win-x86\native\CpuMathNative.dll
+
+C:\git\testapps\cpumath>bin\Debug\netcoreapp3.0\cpumath.exe
+Hello World!
+```
+
+Notice the inclusion of native dependencies for Linux, macOS and two Windows architectures. At the same time, the application includes an executable that will only run on Windows x64. If the application is launched with `dotnet cpumath.dll`, it will then run on all of the architectures supported by the native dependencies.
+
+## Proposed Behavior
+
+`dotnet build` should have three modes:
+
+* Architecture-agnostic, framework-dependent, application that includes native dependencies for all supported architectures and no EXE
+* Achitecture-specific, framework-dependent, application that includes an EXE and native dependencies for a single targeted architecture
+* Architecture-specific, self-contained, application that includes EXE and native dependencies for a single targeted architecture
+
+The proposal is:
+
+* Retain the first behavior as the default.
+* Enable the second behavior as opt-in. Enable this behavior in Windows client application templates.
+* Maintain the third behavior as opt-in (change defaults for `-r` and `--self-contained` for 3.0+).
+
+The first behavior is most similar to the .NET Core 2.2 behavior. It is a good behavior because it builds an application with no assumptions and produces artifacts with the broadest scope and utility. Many developers will want the second behavior. It needs to be very easy to opt-in to that behavior, either via the CLI or with a single property in a template.
```

Comment:
Maybe clarify that the template is doing a simple change in the `csproj` (if I understand you), because users can also do that to existing apps.

Reply:
You are right. Template wasn't the right term.